docs/keepalived.md

```diff
@@ -0,0 +1,96 @@
+# Purpose of keepalived implementation
+
+Adding support for [keepalive](http://www.keepalived.org/documentation.html) to enable high availability in an haproxy deployment, leveraging the https://en.wikipedia.org/wiki/Virtual_Router_Redundancy_Protocol more formally [RFC 5798](https://tools.ietf.org/html/rfc5798). See [keep-alived man page](https://linux.die.net/man/5/keepalived.conf) for more precision over capabilities of keep-alived as well as the [keep-alived user manual](http://www.keepalived.org/pdf/UserGuide.pdf)
+
+This enables declaring an virtual IP (``keepalived.vip``) that will automatically fail over between the multiple haproxy VMs: the master will initially be the bosh vm for haproxy job instance 0. The default IP addresses assigned by bosh to vms on eth0 are used within the VRRP protocol.
+
+Prereqs:
+ * The haproxy VMs must be within the same broadcast domain, i.e. receive multicast traffic sent to the 224.0.0.18 broadcast and IP protocol number 112.
+* The clients using this VIP must be within the [same broadcast domain](https://en.wikipedia.org/wiki/Broadcast_domain) as the haproxy vms and accepting ARP gratuitious
+
```

Comment:
Another potential pre-req - does the IaaS need to support traffic from multiple IPs originating from the same host (see the `source_dest_check` cloud property for AWS: https://bosh.io/docs/aws-cpi.html)?

Not sure how that applies to other infrastructures and their networking layers. Some might be fine, some might have issues. Not necessarily a bad thing, but something to make people aware of?

Also, do users need to pick a VIP for haproxy, and reserve that in BOSH networking, so BOSH doesn't attempt to dole out the address to VMs with dynamic IP addressing?